src/core/sock.h

```diff
@@ -78,6 +79,9 @@ struct nn_sock
     /*  Next endpoint ID to assign to a new endpoint. */
     int eid;
 
+    /*  Count of active holds against the socket. */
+    int holds;
```

Comment:
I don't think atomic adds anything. I've been really careful to use this only under the ctx lock.  If anything adding an atomic may be misleading by drawing attention away from the real protection.